### ISSUE TITLE
when a route's component has nested component using the boards api, the api needs to be initialized twice

### ISSUE THREAD

**Issue by fazo96** [fazo96/ipfs-boards#45]
``` javascript
// When a component inside the component being rendered by the router also needs
// access to the boards api, it appears unitialized and never initializes to it
// for no apparent reason. Calling init twice (one automgically and one
// when the root component mounts) works as a cheap, horrible workaround
```

Possible fixes:
- use react `props` to pass down access to the api and solve #52

**Comment by fazo96** [comment]
Fixed by #52 I guess. Will reopen if issue comes back

**Comment by fazo96** [comment]
It's back. I think I know how to solve it this time:

non-toplevel Components need to reinitialize or rerender when the "init" event is emitted by the API
